### DISCUSSION TITLE
Fetch Changelog from Subdirectory

### DISCUSSION THREAD

**Discussion by mogrady88** [renovatebot/renovate#28908] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I have several projects that are consumed by other teams. In these projects I would like to set the correct path to my changelog so that the renovate MRs that come up for people will have the pertinent changelog entries in the MR. Some projects are working correctly when the CHANGELOG is adjacent to the renovate.json file and the git url is set correctly. My problem is with a single project that contains several subdirectories like so:

```
// my project
renovate.json
package.json
gitlab-ci.yml
README
src/
   subproject-1/
       package.json
       src/
         ...somecode
   subproject-2/
       package.json
       src/
         ...somecode
   subproject-3/
       package.json
       src/
         ...somecode
   self-hosted-docs-for-all-projects/
       package.json
       CHANGELOG // contains updates for project 1,2,3 and I would like to point my root config here
       src/
         ...somecode
  ``` 
  The changelog I'd like to point to should be formatted just fine, I just need to make renovate aware of it. Its a single git repository at the root with some sub projects and a documentation app (where we keep the CHANGELOG as we have a page for it on the docs app)

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
npm has a standard way for packages to indicate this using the `directory` field under `repository`, e.g.

```
{
  "repository": {
    "type": "git",
    "url": "https://github.com/facebook/react.git",
    "directory": "packages/react-dom"
  }
}
```

**Reply by mogrady88** [reply]
I will give that a try. Thank you!

**Comment by robin-ln** [comment]
Hello!

I have the same issue with [my repository](https://gitlab.com/robin.louarn/monorepo-test).

I've configured my `package.json` here: [packages/cli-create-app/package.json](https://gitlab.com/robin.louarn/monorepo-test/-/blob/main/packages/cli-create-app/package.json?ref_type=heads).

I've published my module on GitLab and added this configuration:

```json
"packageRules": [
  {
    "matchPackageNames": [
      "...",
      "@my-company/cli-create-app"
    ],
    "registryUrls": [
      "https://gitlab.com/api/v4/projects/56880161/packages/npm/"
    ]
  }
]
```

However, the changelogs are not showing up in the section `Release Notes` of the [MR](https://gitlab.com/robin.louarn/monorepo-test/-/merge_requests/47).

Could you help me with this? 🙏
